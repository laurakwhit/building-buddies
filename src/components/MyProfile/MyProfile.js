import React, { Component } from 'react';
import { getUserInterests } from '../../utilities/userApiCalls';
import Nav from '../Nav/Nav';

import './MyProfile.scss';

class MyProfile extends Component {
  state = {
    userInterests: []
  }

  async componentDidMount() {
    const userInterests = await getUserInterests(this.props.currentUser.id);
    this.setState(userInterests);
  }

  render() {
    const { currentUser, interests } = this.props;
    const { userInterests } = this.state;
    console.log(interests)
    const displayedInterests = interests.map((interest, index) => {
      const checked = userInterests.includes(interest.name);
      return <div key={index}>
        <input type="checkbox" name={interest.name} value={interest.name} checked={checked} onClick={this.updateUserInterests}/> 
        <label htmlFor={interest.name}>{interest.name}</label>
      </div>
    })
    return (
      <div className="my-profile">
        <Nav />
        <div className="main-content">
          <img src={require('../../assets/profile.jpg')} alt='profile' />
          <h2>{currentUser.name}</h2>
          <article>
            Humblebrag shabby chic gochujang coloring book, squid XOXO brunch tumeric
            selvage hammock vape. Lomo lumbersexual glossier, wayfarers skateboard irony
            humblebrag raclette taxidermy ennui freegan af craft beer PBR&B meh. Asymmetrical
            thundercats intelligentsia, lo-fi swag fingerstache palo santo skateboard
            flexitarian lyft selvage vice poke gluten-free. Kombucha ramps chambray street
            art messenger bag.
          </article>
          <div>
            {displayedInterests}
          </div>
        </div>
      </div>
    );
  }
}

export default MyProfile;
