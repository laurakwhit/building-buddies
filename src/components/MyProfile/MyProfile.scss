.my-profile {
  display: flex;

  .main-content {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: calc(100% - 20rem);
  }

  .basic-info {
    display: flex;
    width: 90%;
    margin-top: 3rem;
  }

  .profile-image {
    width: 200px;
    height: 200px;
    border-radius: 50%;
  }

  .user-info {
    display: flex;
    flex-direction: column;
    align-items: left;
    padding-left: 30px;
    text-align: left;

    h2 {
      margin-top: 0;
    }
  }

  .profile-interests {
    display: grid;
    grid-template-columns: repeat(4, 1fr);
    justify-items: start;
    grid-gap: 15px;
    margin-top: 5rem;
    padding: 2rem;

    border-radius: 10px;
    color: #fff;
    background: #0f2027; /* fallback for old browsers */
    background: -webkit-linear-gradient(
      to right,
      #2c5364,
      #203a43,
      #0f2027
    ); /* Chrome 10-25, Safari 5.1-6 */
    background: linear-gradient(
      to right,
      #2c5364,
      #203a43,
      #0f2027
    ); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */

    -webkit-box-shadow: 11px 10px 36px -12px rgba(0, 0, 0, 0.66);
    -moz-box-shadow: 11px 10px 36px -12px rgba(0, 0, 0, 0.66);
    box-shadow: 11px 10px 36px -12px rgba(0, 0, 0, 0.66);
  }

  label {
    font-size: 1.2rem;
    margin-left: 0.5rem;
  }
}
