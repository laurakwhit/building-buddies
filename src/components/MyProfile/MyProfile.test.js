import React from 'react';
import { shallow } from 'enzyme';

import MyProfile from './MyProfile';
import {
  mockInterests,
  mockUser,
  mockUserInterests
} from '../../utilities/mockData';

describe('MyProfile component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <MyProfile
        currentUser={mockUser}
        interests={mockInterests}
        userInterests={mockUserInterests}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
