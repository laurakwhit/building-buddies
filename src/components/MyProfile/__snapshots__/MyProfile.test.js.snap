// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MyProfile />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Nav />,
        <div
          className="main-content"
        >
          <img
            alt="profile picture"
            src="profile.jpg"
          />
          <h2>
            Name
          </h2>
          <article>
            Humblebrag shabby chic gochujang coloring book, squid XOXO brunch tumeric selvage hammock vape. Lomo lumbersexual glossier, wayfarers skateboard irony humblebrag raclette taxidermy ennui freegan af craft beer PBR&B meh. Asymmetrical thundercats intelligentsia, lo-fi swag fingerstache palo santo skateboard flexitarian lyft selvage vice poke gluten-free. Kombucha ramps chambray street art messenger bag.
          </article>
          <div>
            <h4>
              interest
            </h4>
          </div>
        </div>,
      ],
      "className": "my-profile",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="profile picture"
              src="profile.jpg"
            />,
            <h2>
              Name
            </h2>,
            <article>
              Humblebrag shabby chic gochujang coloring book, squid XOXO brunch tumeric selvage hammock vape. Lomo lumbersexual glossier, wayfarers skateboard irony humblebrag raclette taxidermy ennui freegan af craft beer PBR&B meh. Asymmetrical thundercats intelligentsia, lo-fi swag fingerstache palo santo skateboard flexitarian lyft selvage vice poke gluten-free. Kombucha ramps chambray street art messenger bag.
            </article>,
            <div>
              <h4>
                interest
              </h4>
            </div>,
          ],
          "className": "main-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "profile picture",
              "src": "profile.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Name",
            },
            "ref": null,
            "rendered": "Name",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Humblebrag shabby chic gochujang coloring book, squid XOXO brunch tumeric selvage hammock vape. Lomo lumbersexual glossier, wayfarers skateboard irony humblebrag raclette taxidermy ennui freegan af craft beer PBR&B meh. Asymmetrical thundercats intelligentsia, lo-fi swag fingerstache palo santo skateboard flexitarian lyft selvage vice poke gluten-free. Kombucha ramps chambray street art messenger bag.",
            },
            "ref": null,
            "rendered": "Humblebrag shabby chic gochujang coloring book, squid XOXO brunch tumeric selvage hammock vape. Lomo lumbersexual glossier, wayfarers skateboard irony humblebrag raclette taxidermy ennui freegan af craft beer PBR&B meh. Asymmetrical thundercats intelligentsia, lo-fi swag fingerstache palo santo skateboard flexitarian lyft selvage vice poke gluten-free. Kombucha ramps chambray street art messenger bag.",
            "type": "article",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h4>
                interest
              </h4>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "interest",
              },
              "ref": null,
              "rendered": "interest",
              "type": "h4",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Nav />,
          <div
            className="main-content"
          >
            <img
              alt="profile picture"
              src="profile.jpg"
            />
            <h2>
              Name
            </h2>
            <article>
              Humblebrag shabby chic gochujang coloring book, squid XOXO brunch tumeric selvage hammock vape. Lomo lumbersexual glossier, wayfarers skateboard irony humblebrag raclette taxidermy ennui freegan af craft beer PBR&B meh. Asymmetrical thundercats intelligentsia, lo-fi swag fingerstache palo santo skateboard flexitarian lyft selvage vice poke gluten-free. Kombucha ramps chambray street art messenger bag.
            </article>
            <div>
              <h4>
                interest
              </h4>
            </div>
          </div>,
        ],
        "className": "my-profile",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="profile picture"
                src="profile.jpg"
              />,
              <h2>
                Name
              </h2>,
              <article>
                Humblebrag shabby chic gochujang coloring book, squid XOXO brunch tumeric selvage hammock vape. Lomo lumbersexual glossier, wayfarers skateboard irony humblebrag raclette taxidermy ennui freegan af craft beer PBR&B meh. Asymmetrical thundercats intelligentsia, lo-fi swag fingerstache palo santo skateboard flexitarian lyft selvage vice poke gluten-free. Kombucha ramps chambray street art messenger bag.
              </article>,
              <div>
                <h4>
                  interest
                </h4>
              </div>,
            ],
            "className": "main-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "profile picture",
                "src": "profile.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Name",
              },
              "ref": null,
              "rendered": "Name",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Humblebrag shabby chic gochujang coloring book, squid XOXO brunch tumeric selvage hammock vape. Lomo lumbersexual glossier, wayfarers skateboard irony humblebrag raclette taxidermy ennui freegan af craft beer PBR&B meh. Asymmetrical thundercats intelligentsia, lo-fi swag fingerstache palo santo skateboard flexitarian lyft selvage vice poke gluten-free. Kombucha ramps chambray street art messenger bag.",
              },
              "ref": null,
              "rendered": "Humblebrag shabby chic gochujang coloring book, squid XOXO brunch tumeric selvage hammock vape. Lomo lumbersexual glossier, wayfarers skateboard irony humblebrag raclette taxidermy ennui freegan af craft beer PBR&B meh. Asymmetrical thundercats intelligentsia, lo-fi swag fingerstache palo santo skateboard flexitarian lyft selvage vice poke gluten-free. Kombucha ramps chambray street art messenger bag.",
              "type": "article",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h4>
                  interest
                </h4>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "interest",
                },
                "ref": null,
                "rendered": "interest",
                "type": "h4",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
